src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1781,7 +1781,7 @@ private string TrimWhiteSpaceHelper(TrimType trimType)
             // Trim specified characters.
             if ((trimType & TrimType.Head) != 0)
             {
-                for (start = 0; start < Length; start++)
+                for (; start < Length; start++)
```

Comment:
Can you please verify that this is not introducing performance regression?

The original loop had canonical form that is recognized by the JIT and that should allow bounds-check elimination for `this[start]`. The loop is not in the canonical form after your change that may prevent the bounds-check elimination optimization from kicking in.

Reply:
Have you verified that there is no regression? What does not code looks like before / after this change?